stock_quant_package_restrict_disposable_reuse/models/stock_move_line.py

```diff
@@ -0,0 +1,45 @@
+# 2024 Copyright ForgeFlow, S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    result_package_id = fields.Many2one(
+        domain="['|', '|', ('location_id', '=', False), ('location_id', '=', location_dest_id),"
+        " ('id', '=', package_id), ('disposable_used', '=', False)]"
+    )
+    package_id = fields.Many2one(
+        domain="[('location_id', '=', location_id), ('disposable_used', '=', False)]"
+    )
+
+    def _check_disposable_package(self, package_id):
+        package = self.env["stock.quant.package"].browse(package_id)
+        if package.package_use == "disposable" and package.disposable_used:
+            raise ValidationError(
+                _(
+                    "You can't reuse a disposable package that has already been used"
+                    " in a validated delivery."
+                )
+            )
```

Comment:
you could have this as an api.constrain and you won't need to inherit the create and write